### PR TITLE
Update - triple store

### PR DESCRIPTION
**Pull request by henrieglesorotos**
**What**

Updating Skosmos database backend from Fuseki -> Virtuoso. This is done to unify the triple store solution we use across Edge, and improve out of the box performance. 

**How**

- docker-compose.yml - remove fuseki, replace with virutoso
- rename fuseki-cache to cache 
- additional `VirtuosoSparql.php` sparql model - referenced in config-docker.ttl.

### CODE REVIEW

**Comment by henrieglesorotos** [comment]
http should be fixed now @namedgraph ?

**Comment by namedgraph** [comment]
Looks like. The server name shouldn't make a difference.

You sure Skosmos supports `skosmos:sparqlDialect "Virtuoso"`?

**Comment by henrieglesorotos** [comment]
Latest erros - 

```
skosmos-web | [Wed Mar 08 21:21:29.086454 2023] [php7:warn] [pid 12] [client 172.22.0.1:55216] PHP Warning:  Declaration of VirtuosoSparql::generateAlphabeticalListQuery($letter, $lang, $limit, $offset, $classes, $showDeprecated = false) should be compatible with GenericSparql::generateAlphabeticalListQuery($letter, $lang, $limit, $offset, $classes, $showDeprecated = false, $qualifier = NULL) in /var/www/html/model/sparql/VirtuosoSparql.php on line 23, referer: http://localhost:9090/skills/en/

skosmos-web | 172.22.0.1 - - [08/Mar/2023:21:21:29 +0000] "GET /rest/v1/skills/labelStatistics?lang=en HTTP/1.1" 500 261 "http://localhost:9090/skills/en/" "Mozilla/5.0 (Macintosh; Intel Mac OS X 10_15_7) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/110.0.0.0 Safari/537.36"

skosmos-web | [Wed Mar 08 21:21:29.098545 2023] [php7:error] [pid 12] [client 172.22.0.1:55216] PHP Fatal error:  Uncaught EasyRdf\\Http\\Exception: HTTP request for SPARQL query failed in /var/www/html/vendor/sweetrdf/easyrdf/lib/Sparql/Client.php:274\nStack trace:\n#0 /var/www/html/vendor/sweetrdf/easyrdf/lib/Sparql/Client.php(131): EasyRdf\\Sparql\\Client->request()\n#1 /var/www/html/model/sparql/GenericSparql.php(93): EasyRdf\\Sparql\\Client->query()\n#2 /var/www/html/model/sparql/GenericSparql.php(1407): GenericSparql->query()\n#3 /var/www/html/model/Vocabulary.php(522): GenericSparql->queryFirstCharacters()\n#4 /var/www/html/controller/WebController.php(478): Vocabulary->getAlphabet()\n#5 /var/www/html/index.php(81): WebController->invokeAlphabeticalIndex()\n#6 {main}\n  thrown in /var/www/html/vendor/sweetrdf/easyrdf/lib/Sparql/Client.php on line 274, referer: http://localhost:9090/skills/en/
```

**Comment by henrieglesorotos** [comment]
> 

I mean, I'm not 100% sure. Looking at this - https://groups.google.com/g/skosmos-users/c/OsAFT_OINb8/m/aHwf_fx4BgAJ. This was way before the repo was dockerised tho.

**Comment by namedgraph** [comment]
Have you tried:
> "Generic" for plain SPARQL 1.1`

**Comment by henrieglesorotos** [comment]
> Have you tried:
> 
> > "Generic" for plain SPARQL 1.1`

That's what I'm trying now. Suggests here is the issue - https://github.com/BeameryEdge/Skosmos/blob/update/triple-store/model/sparql/GenericSparql.php#L1407

**Comment by namedgraph** [comment]
No idea if related but I was looking at this at some point: https://github.com/NatLibFi/Skosmos/issues/632

Virtuoso could well be buggy.

**Comment by namedgraph** [comment]
Also this: https://groups.google.com/g/skosmos-users/c/11ROK0C7NyU/m/iGv5mL3hCgAJ

**Comment by henrieglesorotos** [comment]
Just noticed I need to change the sparql endpoint in the vocab configs too. Trying that.

**Comment by namedgraph** [comment]
Delete `dockerfiles/config/skosmos.ttl`? It's Fuseki-specific.

**Comment by henrieglesorotos** [comment]
> Delete `dockerfiles/config/skosmos.ttl`? It's Fuseki-specific.

Sure - Will give that a go.

**Review by namedgraph — CHANGES_REQUESTED** [review]

File: dockerfiles/docker-compose.yml [hunk 1]
```diff
@@ -1,24 +1,24 @@
...
+    container_name: virtuoso
+    hostname: virtuoso
     user: root
-    image: atomgraph/fuseki
-    ports:
-      - 9030:3030
+    image: tenforce/virtuoso:1.3.1-virtuoso7.2.2
     environment:
-      - JAVA_OPTIONS=-Xmx16G -Xms16G
+      SPARQL_UPDATE: "true"
```
[the remark this answers is not in the document]
Comment (namedgraph) [reply dockerfiles/docker-compose.yml:10]: Or just remove the `SPARQL_UPDATE` setting altogether to make the endpoint read-only

**Review by namedgraph — APPROVED** [review]

**Pull request merged**